src/compiler/mir/ir.rs

```diff
@@ -242,6 +263,37 @@ impl ScopeId {
     }
 }
 
+/// An argument for a function.  These are always immutable and are always
+/// in the root scope: therefore, there is no scope or mutable property.
+#[derive(Debug, PartialEq, Clone)]
+pub struct ArgDecl {
+    /// Name of this variable
+    name: StringId,
+    /// The type of this variable
+    ty: Type,
+    /// The span of code where this variable was declared
+    span: Span,
+}
+
+impl ArgDecl {
+    pub fn new(name: StringId, ty: &Type, span: Span) -> ArgDecl {
```

Comment:
```suggestion
    fn new(name: StringId, ty: &Type, span: Span) -> ArgDecl {
```

Reply:
This should not be public, only the Procedure should be able to create a new ArgDecl.